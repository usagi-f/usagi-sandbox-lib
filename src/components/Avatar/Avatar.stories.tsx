import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Avatar } from './Avatar';

storiesOf('Avatar', module)
  .add('Simple render', () => <Avatar />)
  .add('with alt', () => <Avatar alt="Custom Alt" />)
  .add('with size', () => (
    <div>
      <Avatar size={12} />
      <Avatar size={24} />
      <Avatar size={48} />
      <Avatar size={64} />
    </div>
  ));
