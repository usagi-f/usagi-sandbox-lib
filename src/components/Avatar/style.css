.avatar {
  border-radius: 50%;
  display: inline-block;
  overflow: hidden;
  vertical-align: top;
}

.avatar img {
  vertical-align: middle;
}
