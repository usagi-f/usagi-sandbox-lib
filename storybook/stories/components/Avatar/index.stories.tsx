import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { Avatar as AvatarComponent, IAvatarProps } from 'usagi-sandbox-lib';
import 'usagi-sandbox-lib/dist/css/all.css';

export const Avatar: React.FC<IAvatarProps> = props => <AvatarComponent {...props} />;

storiesOf('Avatar', module)
  .add('Simple render', () => <Avatar />)
  .add('with alt', () => <Avatar alt="Custom Alt" />)
  .add('with size', () => (
    <div>
      <Avatar size={12} />
      <Avatar size={24} />
      <Avatar size={48} />
      <Avatar size={64} />
    </div>
  ));
