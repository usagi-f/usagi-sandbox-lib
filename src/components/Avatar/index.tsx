import * as React from 'react'
import * as style from './style.css'

interface IAvatarProps extends React.Props<{}> {
  alt?: string;
  size?: number;
}

const Avatar: React.FC<IAvatarProps> = (props: IAvatarProps) => (
  <div className={style.avatar}>
    <img src={`https://via.placeholder.com/${props.size}`} alt={props.alt} width={props.size} />
  </div>
)

Avatar.defaultProps = {
  alt: '',
  size: 32,
}

export {
  Avatar,
  IAvatarProps,
}
