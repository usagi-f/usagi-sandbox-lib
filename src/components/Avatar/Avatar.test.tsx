import * as React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Avatar, IAvatarProps } from './Avatar';

test('Rendered by props', () => {
  const props: IAvatarProps = {
    alt: 'foo',
    size: 10,
  };
  const wrapper = shallow(<Avatar {...props} />);
  expect(wrapper.find('img').prop('alt')).toBe(props.alt);
  expect(wrapper.find('img').prop('width')).toBe(props.size);
});

test('Snapshot', () => {
  const props: IAvatarProps = {
    alt: 'foo',
    size: 10,
  };
  const tree = shallow(<Avatar {...props} />);
  expect(toJson(tree)).toMatchSnapshot();
});
