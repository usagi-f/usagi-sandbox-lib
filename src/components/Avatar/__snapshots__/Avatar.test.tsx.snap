// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <img
    alt="foo"
    src="https://via.placeholder.com/10"
    width={10}
  />
</div>
`;
